esti/import_test.go

```diff
@@ -191,3 +195,52 @@ func testImport(t *testing.T, ctx context.Context, repoName string, importPath s
 	require.NoError(t, err, "failed to commit")
 	require.Equal(t, http.StatusCreated, commitResp.StatusCode())
 }
+
+func TestAzureDataLakeV2(t *testing.T) {
+	importPrefix := viper.GetString("adls_import_base_url")
+	if importPrefix == "" {
+		t.Skip("No Azure data lake storage path prefix was given")
+	}
+
+	ctx, _, repoName := setupTest(t)
+	defer tearDownTest(repoName)
+
+	tests := []struct {
+		name         string
+		filesToCheck []string
+	}{
+		{
+			name:         "empty-folders",
+			filesToCheck: []string{},
+		},
+		{
+			name: "prefix-item-order",
+			filesToCheck: []string{
+				"aaa",
+				"helloworld.csv",
+				"zero",
+				"helloworld/myfile.csv",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		importBranch := fmt.Sprintf("%s-%s", importBranchBase, tt.name)
+		// each test is a folder under the prefix import
+		t.Run(tt.name, func(t *testing.T) {
+			importPath, err := url.JoinPath(importPrefix, tt.name)
+			if err != nil {
+				t.Fatal("Import URL", err)
+			}
+			testImport(t, ctx, repoName, importPath, importBranch)
+			if len(tt.filesToCheck) == 0 {
+				resp, err := client.ListObjectsWithResponse(ctx, repoName, importBranch, &api.ListObjectsParams{})
+				require.NoError(t, err)
+				require.NotNil(t, resp.JSON200)
+				require.Empty(t, resp.JSON200.Results)
+			} else {
+				verifyImportObjects(t, ctx, repoName, importTargetPrefix, importBranch, tt.filesToCheck, 0)
```

Comment:
Can we also check that the ordering of a List request returned from the server is of correct order? If pebble won't fail we might still have weird stuff in the range.